lib/main_common.pm

```diff
@@ -2827,48 +2827,72 @@ sub load_extra_tests_kernel {
     loadtest "kernel/tuned";
 }
 
-sub load_publiccloud_tests {
-    if (get_var('PUBLIC_CLOUD_PREPARE_TOOLS')) {
-        loadtest "publiccloud/prepare_tools";
-    }
-    elsif (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS') && !get_var('PUBLIC_CLOUD_QAM')) {
-        loadtest "publiccloud/img_proof";
-    }
-    elsif (get_var('PUBLIC_CLOUD_QAM')) {
-        loadtest "publiccloud/download_repos";
-        my $args = OpenQA::Test::RunArgs->new();
-        loadtest "publiccloud/ssh_interactive_init", run_args => $args;
-        loadtest "publiccloud/register_system",      run_args => $args;
-        loadtest "publiccloud/transfer_repos",       run_args => $args;
-        loadtest "publiccloud/patch_and_reboot",     run_args => $args;
-        if (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
-            loadtest("publiccloud/img_proof", run_args => $args);
-        } else {
-            loadtest "publiccloud/ssh_interactive_start", run_args => $args;
-            loadtest "publiccloud/instance_overview" unless get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS');
+sub load_qam_publiccloud_tests {
+    my $args = OpenQA::Test::RunArgs->new();
+
+    loadtest "publiccloud/download_repos";
+    loadtest "publiccloud/ssh_interactive_init", run_args => $args;
+    loadtest "publiccloud/register_system",      run_args => $args;
+    loadtest "publiccloud/transfer_repos",       run_args => $args;
+    loadtest "publiccloud/patch_and_reboot",     run_args => $args;
+    if (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
+        loadtest("publiccloud/img_proof", run_args => $args);
+        return;    # Finalize (i.e. "terraform destroy") is handled internally
+    } elsif (get_var('PUBLIC_CLOUD_LTP')) {
+        loadtest('publiccloud/run_ltp', run_args => $args);
+        return;    # Finalize (i.e. "terraform destroy") is handled internally
+    } elsif (get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME')) {
+        loadtest('publiccloud/boottime', run_args => $args);
+        return;    # Finalize (i.e. "terraform destroy") is handled internally
+    } elsif (get_var('PUBLIC_CLOUD_FIO')) {
+        loadtest('publiccloud/storage_perf', run_args => $args);
+        return;    # Finalize (i.e. "terraform destroy") is handled internally
+    } else {
+        loadtest "publiccloud/ssh_interactive_start", run_args => $args;
+        loadtest "publiccloud/instance_overview" unless get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS');
+
+        if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS')) {
             load_extra_tests_prepare();
-            load_extra_tests_docker()  if (get_var('PUBLIC_CLOUD_CONTAINERS'));
-            load_extra_tests_console() if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS'));
-            loadtest "publiccloud/ssh_interactive_end", run_args => $args;
+            load_extra_tests_console();
+        }
+        if (get_var('PUBLIC_CLOUD_CONTAINERS')) {
+            load_extra_tests_prepare();
+            load_extra_tests_console();
         }
```

Comment:
This is used nowhere so there is no reason to update this.
I suggest to keep this untouched and remove it in next PR.

Reply:
Or please ignore this. We can fix it later, properly.